plugins/org.eclipse.fordiac.ide.application/src/org/eclipse/fordiac/ide/application/editors/SubAppNetworkEditor.java

```diff
@@ -73,8 +73,18 @@ public void dispose() {
 
 	@Override
 	public void setInput(final IEditorInput input) {
-		if (input instanceof final SubApplicationEditorInput subAppInput) {
-			setModel(subAppInput.getSubApp().getSubAppNetwork());
+		if (!(input instanceof final SubApplicationEditorInput subAppEI)) {
+			throw new IllegalArgumentException(
+					"SuppApp editors only accept SubApplicationEditorInput as valid inputs!"); //$NON-NLS-1$
+		}
+		final SubApplicationEditorInput currentEditorInput = (SubApplicationEditorInput) getEditorInput();
+		if (currentEditorInput != null && currentEditorInput.getContent() != subAppEI.getContent()) {
+			throw new IllegalArgumentException(
+					"Editor input with new content given to subapp editor. This is currently not supported!"); //$NON-NLS-1$
+		}
```

Comment:
I believe this check is now redundant with `DiagramEditorWithFlyoutPalette.setInput(IEditorInput)`?